lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -582,8 +583,17 @@ def __init__(
               only raise ValueError.
             refresh_event: an optional bound event or list of bound events which
                 will be observed to re-apply the patch.
+            allow_patching_charm_container: Whether to allow patching the "charm" container.
         """
         super().__init__(charm, "{}_{}".format(self.__class__.__name__, container_name))
+
+        if container_name == "charm" and not allow_patching_charm_container:
```

Comment:
To act on `allow_patching_charm_container` we'd need a `force`?
In any case, I think we shouldn't act on `charm`.

Reply:
I'm not saying we should do force in any way, just let the user try if they know better (e.g they know they are on a juju version where this isn't a problem)